### DISCUSSION TITLE
Custom Login Form Error : Redirecting to https://localhost:3000/api/auth/signin?csrf=true using CredentialsProvider

### DISCUSSION THREAD

**Discussion by devb** [nextauthjs/next-auth#4318] [Help]
**DESCRIPTION:**
I'm facing while implementing custom login form in my nextjs site using next-auth.
It keeps on redirecting to "https://localhost:3000/api/auth/signin?csrf=true"

**SERVER SIDE CODE WRITTEN IN "[...nextauth.js]" FILE INSIDE "pages/api/auth/" folder.**
```
import NextAuth from 'next-auth';
import CredentialsProvider from 'next-auth/providers/credentials';

const providers = [
    CredentialsProvider({
        id: 'credentials',
        name: 'credentials',
        authorize: async (credentials) => {
            console.log("AUTHORIZING...",credentials,req);
            const res = await fetch(process.env.apiURL+"api/users/authenticate", {
                method: 'POST',
                body: JSON.stringify(credentials),
                headers: { "Content-Type": "application/json" }
            });
            const user = await res.json();
            if (res.ok && user) {
                return user;
            }
            return null;
        }
    }),
];

const callbacks = {
    async signIn({ user, account, profile, email, credentials }) {
        console.log("IN CALLBACK HERE!!!");
        const isAllowedToSignIn = true
        if (isAllowedToSignIn) {
          return true
        } else {
          return false
        }
    },
    async redirect({ url, baseUrl }) {
        return baseUrl;
    },
    async session(session, token) {
      console.log("IN THE CALLBACK SESSION NOW");
      session.accessToken = token.accessToken
      return session
    }
}

const options = {
    providers,
    callbacks,
    secret: process.env.JWT_SECRET,
    pages: {
        signIn: '/my-account',
        signUp: '/signup',
        error: '/my-account'
    },
    debug: true
}

export default (req, res) => NextAuth(req, res, options);
```

**MY CUSTOM LOGIN FORM'S SUBMIT BUTTON ACTION:**
```
const res = await signIn('credentials', { redirect: false, username: user.username, passw: user.passw, callbackUrl: `${window.location.origin}` });
            console.log(res);
            if (res?.error) {
                console.error(res.error);
            }
            if (res.url) router.push(res.url);
```

**ENVIRONMENT VARIABLES**
NEXTAUTH_URL=https://localhost:3000/api/auth
NEXTAUTH_SECRET=PETSFINEST

**ERROR IS**
the Authorize function of Credentials Provider is not at all being called.. as soon as I press SIGNIN button from my custom login form, it simply redirects to "https://localhost:3000/api/auth/signin?csrf=true" URL and thats it.

Please help guys.. what am I doing wrong here? There's no other details available anywhere, and I've followed the documentation provided on next-auth site.

**Comment by devb** [comment]
No one is facing this issue ???!!

**Reply by hensansi** [reply]
If you post a repo on github I can have a look at it, after having a look at it nothing seems blatantly wrong.

**Comment by fromwhite** [comment]
hey guys
I hit the same issue,long time of headache.
In my case,updating node v20.7.0 and next-auth v4.13.0,and now everything works as expected.[#2573](https://github.com/nextauthjs/next-auth/discussions/2573)
At the same time, it may bring another issue, using unstable_ Replace getServerSession,[#5187](https://github.com/nextauthjs/next-auth/issues/5187)

There is another case where it is the latest version v4.22.1,but in 996, I didn't have time to test it, my case didn't have any requirements for the version.
If you don't want to reverse the version, perhaps you can take a look at its handling [v4.22.1](https://github.com/vercel/nextjs-postgres-auth-starter)
